src/report.rs

```diff
@@ -72,8 +75,11 @@ impl Reporter {
                 .write()
                 .expect("failed to obtain write lock for stats");
             let mut snap = StatMap::default();
-            for (k, v) in orig.drain() {
-                snap.insert(k, v);
+            for (k, v) in orig.iter_mut() {
+                let value = ::std::mem::replace(v,
+                                                Histogram::<u64>::new(HISTOGRAM_PRECISION)
+                                                    .expect("failed to build Histogram"));
```

Comment:
In fact, by replacing the histogram, we're changing behavior so that histograms never fall out of the registry.  It's by-design that histograms are tracked only when they have values---this will make it easier for us to track things like per-host stats without worrying about tracking an infinite number of stats over time.